apps/vs-code-react/src/webviewCommunication.tsx

```diff
@@ -66,7 +66,7 @@ export const WebViewCommunication: React.FC<{ children: ReactNode }> = ({ childr
       dispatch(initialize(message.data.project));
     }
 
-    switch (projectState?.project) {
+    switch (projectState?.project ?? message?.data?.project) {
```

Comment:
Tested if the state is updated right after the dispatch call on top; not updated immediately, so using the param value on initialize call.